library/ssl_tls13_server.c

```diff
@@ -761,6 +761,9 @@ static int ssl_tls13_determine_key_exchange_mode( mbedtls_ssl_context *ssl )
     else
     if( ssl_tls13_check_ephemeral_key_exchange( ssl ) )
     {
+#if defined(MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED)
+        mbedtls_ssl_remove_psk( ssl );
```

Comment:
In ssl_tls13_keys.c I can see that eventually we check the key exchange mode in `mbedtls_ssl_tls13_key_schedule_stage_early()` to use a PSK to derive the first secret. Thus do we still need this ? If not please just revert or remove the commit " remove psk key when ephemeral selected ".

Reply:
removed